core/trino-spi/src/test/java/io/trino/spi/connector/TestColumnMetadata.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TestColumnMetadata
+{
+    @Test
+    public void testBuilderFrom()
+    {
+        ColumnMetadata originColumnMetadata = ColumnMetadata.builder()
+                .setName("test_column")
+                .setType(INTEGER)
+                .setDefaultValue(Optional.of("1"))
+                .setNullable(false)
+                .setComment(Optional.ofNullable("test_comment"))
+                .setExtraInfo(Optional.of("test_extra_info"))
+                .setHidden(false)
+                .setProperties(ImmutableMap.of("test_key", "test_value"))
+                .build();
```

Comment:
Make `ColumnMetadata` constructor package private, `@VisibleForTesting` and call here.
This would ensure the test stays up to date, should there any new fields be added.